runners/direct-java/src/main/java/org/apache/beam/runners/direct/RootTransformEvaluatorFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.direct;
+
+import java.util.Collection;
+import org.apache.beam.runners.direct.DirectRunner.CommittedBundle;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.transforms.AppliedPTransform;
+import org.apache.beam.sdk.transforms.PTransform;
+
+/**
+ * A {@link TransformEvaluatorFactory} for {@link PTransform PTransforms} that are at the root of a
+ * {@link Pipeline}. Provides a way to get initial inputs, which will cause the {@link PTransform}
+ * to produce all of the appropriate output.
+ */
+interface RootTransformEvaluatorFactory extends TransformEvaluatorFactory {
```

Comment:
It should not. I've split it into an independent interface. However, I'm (generally) keeping the input providers within the Evaluator Factory implementations, as the initial inputs have to match the type of the actual Evaluator.